hydra-plutus/src/Hydra/Contract/Commit.hs

```diff
@@ -107,20 +116,21 @@ validator (_party, _commit, headId) r ctx =
   headOutputValue =
     txOutValue . head $ txInfoOutputs (scriptContextTxInfo ctx)
 
-compiledValidator :: CompiledCode ValidatorType
+compiledValidator :: TypedValidator CommitValidator
 compiledValidator =
-  $$(PlutusTx.compile [||wrap validator||])
+  mkTypedValidator @CommitValidator
+    $$(PlutusTx.compile [||validator||])
+    $$(PlutusTx.compile [||wrap||])
  where
-  wrap = wrapValidator @DatumType @RedeemerType
-
-validatorScript :: SerialisedScript
-validatorScript = serialiseCompiledCode compiledValidator
+  wrap = mkUntypedValidator @ScriptContext @(DatumType CommitValidator) @(RedeemerType CommitValidator)
+validatorHash' :: ValidatorHash
+validatorHash' = Script.validatorHash compiledValidator
 
-validatorHash :: ScriptHash
-validatorHash = scriptValidatorHash PlutusScriptV2 validatorScript
+validatorHashAddress :: Address
+validatorHashAddress = Ledger.scriptValidatorHashAddress validatorHash' Nothing
```

Comment:
I think we do not need `Address` values for our script as we only would need to depend on the script hashes (`ValidatorHash` now?)